src/webui/api/rsscontroller.cpp

```diff
@@ -91,6 +94,30 @@ void RSSController::itemsAction()
     setResult(jsonVal.toObject());
 }
 
+void RSSController::markItemAsReadAction()
+{
+    requireParams({"itemPath"});
+
+    const QString itemPath {params()["itemPath"]};
+    RSS::Item *item = RSS::Session::instance()->itemByPath(itemPath);
+    if (item)
+        item->markAsRead();
+}
+
+void RSSController::markArticleAsReadAction()
```

Comment:
I would implement only one "markAsReadAction" so if "articleId" parameter is passed it will affect the article otherwise the item.